include/revng/Yield/Support/GraphLayout/SugiyamaStyle/InternalGraph.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+#include <variant>
+
+#include "revng/ADT/GenericGraph.h"
+#include "revng/Yield/Support/GraphLayout/Traits.h"
+
+namespace yield::layout::sugiyama {
+
+class InternalGraph;
+
+namespace detail {
+
+struct InternalNode {
+  layout::Point Center;
+  layout::Size Size;
+
+  bool IsVirtual;
+
+private:
+  std::size_t Index;
+
+private:
+  friend InternalGraph;
+  InternalNode(std::size_t Index, layout::Size &&Size) :
+    Center({ 0, 0 }), Size(std::move(Size)), IsVirtual(false), Index(Index) {}
+  InternalNode(std::size_t Index) :
+    Center({ 0, 0 }), Size({ 0, 0 }), IsVirtual(true), Index(Index) {}
+
+public:
+  InternalNode(const InternalNode &) = delete;
```

Comment:
I guess this is non-copyable for ensuring we maintain the unique `Index` right?

Please explicitly `delete` also the copy-assignment for https://en.cppreference.com/w/cpp/language/rule_of_three

Reply:
Done.